graphprot/graphprot_seqmodel

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+
+DESCRIPTION = """
+GraphProt:
+
+Explicit Decomposition with Neighborhood (EDeN) utility program for
+sequence-based modeling of binding preferences of RNA-binding proteins.
+"""
+
+EPILOG = """
+Author: Daniel Maticzka
+Copyright: 2015
+License: GPL
+Maintainer: Daniel Maticzka
+Email: maticzkd@informatik.uni-freiburg.de
+Status: Development
+
+Please cite:
+
+    Daniel Maticzka, Sita J. Lange, Fabrizio Costa, Rolf Backofen
+    GraphProt: modeling binding preferences of RNA-binding proteins
+    Genome Biology, 2014, 15(1), R17
+"""
+
+import os
+import logging
+
+from numpy.random import randint, uniform
+from sklearn.linear_model import SGDClassifier
+from eden.graph import Vectorizer
+from eden.model_base import ModelInitializerBase, main_script
+
+
+class ModelInitializer(ModelInitializerBase):
+
+    def _load_data(self, input_file):
+        """Load sequence tuples from file."""
+        from eden.converter.fasta import fasta_to_sequence
+        seqs = fasta_to_sequence(input_file)
+        return seqs
+
+    def load_data(self, args):
+        """Load positive class data from file."""
+        return self._load_data(args.input_file)
+
+    def load_positive_data(self, args):
+        """Load positive class data from file."""
+        return self._load_data(args.positive_input_file)
+
+    def load_negative_data(self, args):
+        """Load negative class data from file."""
+        return self._load_data(args.negative_input_file)
+
+    def pre_processor_init(self, n_iter):
+        """Setup conversion of sequence tuples to graphs."""
+        def pre_processor(seqs, **args):
+            """Convert sequence tuples to graphs."""
+            from eden.converter.fasta import sequence_to_eden
+            graphs = sequence_to_eden(seqs)
+            return graphs
+        pre_processor_parameters = {}
+        return pre_processor, pre_processor_parameters
+
+    def vectorizer_init(self, n_iter):
+        """Setup conversion of graphs to features."""
+        vectorizer = Vectorizer()
+        vectorizer_parameters = {
+            'r': [1, 0, 2, 3, 4],
+            'd': [4, 0, 1, 2, 3, 4, 5, 6],
+            'nbits': [14, 16, 18]}
+        return vectorizer, vectorizer_parameters
+
+    def estimator_init(self, n_iter):
+        """Setup the estimator and set of matching parameter choices."""
+        estimator = SGDClassifier(average=True, class_weight='auto', shuffle=True)
+        estimator_parameters = {'n_iter': randint(5, 200, size=n_iter),
```

Comment:
To make it easier for testing it would be nice to be able to put in a fixed seed with andom.seed().
Somewhere, I guess this can be done on a higher level not here in a function.